engine/classes/ElggUser.php

```diff
@@ -667,4 +667,40 @@ public function setPassword($password) {
 		$this->attributes['password'] = "";
 		$this->attributes['password_hash'] = _elgg_services()->passwords->generateHash($password);
 	}
+
+	/**
+	 * Enable or disable a notification delivery method
+	 *
+	 * @param string $method  Method name
+	 * @param bool   $enabled Enabled or disabled
+	 * @return bool
+	 */
+	public function setNotificationSetting($method, $enabled = true) {
```

Comment:
I'm not sure whether it's wise to add any more methods within `ElggUser`. The class is already doing more than it perhaps should. Better would be to have a separate subscriptions service to take care of it:

```php
$subscriptions->setPersonalSetting($user, $method, $enabled);
```

But this is not a blocker. We can always deprecate the function later if we want to separate the feature.